src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -3017,11 +3017,879 @@ partial class C
+            // There is no way in C# to call a setter without passing an argument for the value, so the CallerMemberName effectively does nothing.
+            // It would be reasonable to also warn here about the CallerInfo attribute on implementation,
+            // but this is a corner case that clearly won't work regardless of which part the attribute is on, so it's not a big deal that the warning is missing.
+            // Verify that our checks for caller-info attributes on implementation part parameters behave gracefully here.
+            var source = """
+                using System;
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+
+                partial class C
+                {
+                    public static void Main()
+                    {
+                        var c = new C();
+                        c[1] = "1";
+                    }
+
+                    public partial string this[int x] { set; }
+                    public partial string this[int x]
+                    {
+                        [param: Optional, DefaultParameterValue("0")]
+                        [param: CallerMemberName]
+                        set
+                        {
+                            Console.Write(value);
+                        }
+                    }
+                }
+                """;
+
+            var verifier = CompileAndVerify(source, expectedOutput: "1");
+            verifier.VerifyDiagnostics();
+        }
+
+        [Theory]
+        [InlineData("[AllowNull] ", "")]
+        [InlineData("", "[AllowNull] ")]
+        public void AllowNull(string defAttrs, string implAttrs)
+        {
+            var source = $$"""
+                #nullable enable
+
+                using System;
+                using System.Diagnostics.CodeAnalysis;
+
+                partial class C
+                {
+                    public static void Main()
+                    {
+                        var c = new C();
+                        try
+                        {
+                            _ = c[null]; // no warning
+                        }
+                        catch
+                        {
+                            Console.Write(1);
+                        }
+                    }
+
+                    public partial string this[{{defAttrs}}string s] { get; }
+                    public partial string this[{{implAttrs}}string s]
+                    {
+                        get
+                        {
+                            return s.ToString(); // https://github.com/dotnet/roslyn/issues/73484: missing a warning here
+                        }
+                    }
+                }
+                """;
+
+            var verifier = CompileAndVerify([source, AllowNullAttributeDefinition], expectedOutput: "1");
+            verifier.VerifyDiagnostics();
+        }
```

Comment:
We suppress obsolete diagnostics when within an obsolete member. This may be interesting to test: obsolete attribute on the definition part, then usage of an obsolete API within the implementation part.
Similarly, a well-known attribute like a nullable attribute on the definition part that should affect the nullability validation on the implementation part.

Reply:
> Similarly, a well-known attribute like a nullable attribute on the definition part that should affect the nullability validation on the implementation part.

`PartialPropertiesTests.AllowNull` should show this, except we hit bug #73484.

Reply:
Added test `PartialPropertiesTests.AllowNull_Property` which does show a similar interesting behavior

Reply:
I do think it would be good to have an explicit test for obsolete as well. Another attribute that we should test is `UnmanagedCallersOnly`; I would expect to get an error for trying to put it anywhere here.